### PR TITLE
Added Minimum Cost Path

### LINKED ISSUE #1396: Min Cost Path in 2d-Array
in c,c++,java and python

### PR DESCRIPTION
**Pull request by somya-kapoor**
Fixes #1396 

#### Checklist:
- [x] 4 space indentation.
- [x] Coding conventions are followed.
- [x] Input is taken dynamically.
- [x] Sample Input / Output is added at the end of file.
- [x] Logic Documentation (Comments).
- [x] File names are correct.

#### Changes proposed in this pull request:
- added Min Cost Path in 2d-Array  

#### Languages Used:
- C
- C++
- java
- python

#### Files Added:
- Minimum_Cost_Path.c 
- Minimum_Cost_Path.cpp
- Minimum_Cost_Path.java 
- Minimum_Cost_Path.py

### CODE REVIEW

File: Minimum_Cost_Path/Minimum_Cost_Path.c [hunk 1]
```diff
@@ -0,0 +1,67 @@
...
+    int min;
+    
+    // For invalid left and right query
+    if (left < 0 || right < 0) 
+        return INT_MAX; 
+    else if (left == 0 && right == 0) 
+        return array[left][right]; 
+    else
+    {
+        // Finging path with minimum cost i.e. which way to move down, right and diagonally lower cells
```
Comment (MastersAbh) [inline Minimum_Cost_Path/Minimum_Cost_Path.c]: Typo in finding

File: Minimum_Cost_Path/Minimum_Cost_Path.c [hunk 2]
```diff
@@ -0,0 +1,67 @@
...
+    // For invalid left and right query
+    if (left < 0 || right < 0) 
+        return INT_MAX; 
+    else if (left == 0 && right == 0) 
+        return array[left][right]; 
+    else
+    {
+        // Finging path with minimum cost i.e. which way to move down, right and diagonally lower cells 
+        if (minimumCost(array, left - 1, right - 1) < minimumCost(array, left - 1, right))
+            min = (minimumCost(array, left - 1, right - 1) < minimumCost(array, left, right - 1))?
```
Comment (MastersAbh) [inline Minimum_Cost_Path/Minimum_Cost_Path.c]: space before ?

File: Minimum_Cost_Path/Minimum_Cost_Path.c [hunk 3]
```diff
@@ -0,0 +1,67 @@
...
+} 
+
+int main() 
+{ 
+    scanf("%d", &row);
+    scanf("%d", &col);
+    int array[row][col];
+    for (int i = 0; i < row; i++)
+        for (int j = 0; j < col; j++)
+            scanf("%d",&array[i][j]);
```
Comment (MastersAbh) [inline Minimum_Cost_Path/Minimum_Cost_Path.c]: space after comma

File: Minimum_Cost_Path/Minimum_Cost_Path.c [hunk 4]
```diff
@@ -0,0 +1,67 @@
...
+    int left,right;
```
Comment (MastersAbh) [inline Minimum_Cost_Path/Minimum_Cost_Path.c]: space after comma

File: Minimum_Cost_Path/Minimum_Cost_Path.cpp [hunk 5]
```diff
@@ -0,0 +1,67 @@
...
+    int min;
+    
+    // For invalid left and right query
+    if (left < 0 || right < 0) 
+        return INT_MAX; 
+    else if (left == 0 && right == 0) 
+        return array[left][right]; 
+    else
+    {
+        // Finging path with minimum cost i.e. which way to move down, right and diagonally lower cells
```
Comment (MastersAbh) [inline Minimum_Cost_Path/Minimum_Cost_Path.cpp]: Typo

File: Minimum_Cost_Path/Minimum_Cost_Path.cpp [hunk 6]
```diff
@@ -0,0 +1,67 @@
...
+    // For invalid left and right query
+    if (left < 0 || right < 0) 
+        return INT_MAX; 
+    else if (left == 0 && right == 0) 
+        return array[left][right]; 
+    else
+    {
+        // Finging path with minimum cost i.e. which way to move down, right and diagonally lower cells 
+        if (minimumCost(array, left - 1, right - 1) < minimumCost(array, left - 1, right))
+            min = (minimumCost(array, left - 1, right - 1) < minimumCost(array, left, right - 1))?
```
Comment (MastersAbh) [inline Minimum_Cost_Path/Minimum_Cost_Path.cpp]: space before ?

File: Minimum_Cost_Path/Minimum_Cost_Path.cpp [hunk 7]
```diff
@@ -0,0 +1,67 @@
...
+                    minimumCost(array, left - 1, right - 1) : minimumCost(array, left, right - 1);
+        else 
+            min = (minimumCost(array, left - 1, right) <  minimumCost(array, left, right - 1))?
```
Comment (MastersAbh) [inline Minimum_Cost_Path/Minimum_Cost_Path.cpp]: space before ?

File: Minimum_Cost_Path/Minimum_Cost_Path.java [hunk 8]
```diff
@@ -0,0 +1,63 @@
...
+        // For invalid left and right query
+        if (left < 0 || right < 0) 
+            return  Integer.MAX_VALUE;  
+        else if (left == 0 && right == 0) 
+            return array[left][right]; 
+        else
+        {
+            // Finging path with minimum cost i.e. which way to move down, right and diagonally lower cells 
+            if (minimumCost(array, left - 1, right - 1) < minimumCost(array, left - 1, right))
+                min = (minimumCost(array, left - 1, right - 1) < minimumCost(array, left, right - 1))?
```
Comment (MastersAbh) [inline Minimum_Cost_Path/Minimum_Cost_Path.java]: space before ?

**Comment by somya-kapoor** [comment]
@MastersAbh Done please review

**Comment by somya-kapoor** [comment]
@marwahmanbir  Please review.

**Pull request merged**